Backend_Flower Delivery

EHIS03042 opened the pull request:
Flower Delivery Website Backend - Milestone 1
This repository contains the backend for the Flower Delivery Website, developed as part of Milestone 1. It provides a RESTful API for managing flower products, including creating, reading, updating, and deleting flower entries, with support for image uploads.
Flower Delivery Website Backend - **Milestones 1 & 2**
The project is built incrementally in milestones. **Milestone 1** focused on the Flower product CRUD functionality. **Milestone 2** introduces user authentication and authorization features, including password-based login and Google OAuth, as well as protected routes and JWT token management.
## Presentation Link(Loom Video) Milestone 1
[[Presentation Link](https://www.loom.com/share/01d3878d146f48498fc133eb910dd095?sid=70aa91f3-b4e5-452a-b78b-46056c6b2a0b)]
## Presentation Link(Loom Video) Milestone 2
[[Presentation Link](https://www.loom.com/share/69083b5e6a0140b2baeab91725fa6da4?sid=74ad1d57-125d-4e68-b64d-188528542952)]

## Live Demo(on Render.com)
[[live_demo](https://flower-delivery-website-af2b.onrender.com)]

EHIS03042 committed:
Recreating Repo after irreconcilable commit histories

EHIS03042 committed:
node_modules and .env created

EHIS03042 committed:
Final Edits

EHIS03042 committed:
Repositioning the README file into the backend folder

EHIS03042 committed:
Remove README.md from root directory

EHIS03042 committed:
Create README.md

EHIS03042 committed:
Restoring Lost Files

EHIS03042 committed:
Created User Logic for Milestone 2

EHIS03042 committed:
General Edits and Audits

EHIS03042 committed:
README File Updated with details for Milestone 2

EHIS03042 committed:
Milestone 2 Presentation Link Added

Devsalmasaeed reviewed (approved):
> > Hi @EHIS03042 
> > Your project is complete! There is nothing else to say other than... it's time to merge it!!! Congratulations! 🎉
> > ![Remote Collaboration](https://camo.githubusercontent.com/636f8f83cc9fdf8edaad0e2e31e471335a0431c15559750d6ff98dc325075ca5/68747470733a2f2f6d65646961312e67697068792e636f6d2f6d656469612f674c7539304f4d6a7a346a3368416b4a6b322f67697068792e6769663f6369643d3437303238666138737831326438323478727679387a6f38627133773030663233686f6a31316771717a7573377130662665703d76315f67696673267269643d67697068792e6769662663743d67)
> > ### Highlights
> > 
> > * ✅ **Professional Readme**
> > * ✅ **Good use of Gitflow**
> > * ✅ **UI looks like the Figma design**
> > * ✅ **Good use of HTML semantics**
> > * ✅ **Professional commit messages**
> > * ✅ **Professional PR title**
> > * ✅ **Professional PR description**
> > 
> > ### Optional Suggestions
- please re
> > Every comment with the `[OPTIONAL]` prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.
> > Cheers and Happy coding! 👏👏👏
> > _Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification._

EHIS03042 committed:
Backend-Stripely

This is the  backend for the Flower Delivery Website. The project is built incrementally in milestones. **Milestone 1** focused on the Flower product CRUD functionality. **Milestone 2** introduces user authentication and authorization features including password-based login and Google OAuth, as well as protected routes and JWT token management.

EHIS03042 committed:
backend

Newer Updates

EHIS03042 committed:
Delete backend/backend/config/cloudinary.js remove duplicate

EHIS03042 committed:
Delete backend/backend/controller/flowerController.js rm dupli

EHIS03042 committed:
Delete backend/backend/controller/userController.js rm dupli

EHIS03042 committed:
Delete backend/backend/middleware/upload.js rm dupli

EHIS03042 committed:
Delete backend/backend/models/flowerModel.js rm dupli

EHIS03042 committed:
Delete backend/backend/models/userModel.js rm dupli

EHIS03042 committed:
Delete backend/backend/routes/flowerRoutes.js rm dupli

EHIS03042 committed:
Delete backend/backend/routes/paymentRoutes.js rm dupli

EHIS03042 committed:
Delete backend/backend/routes/userRoutes.js rm dupli

EHIS03042 committed:
Delete backend/backend/scripts/cleanupAndMigrate.js rm dupli

EHIS03042 committed:
Delete backend/backend/scripts/fixImagesHybrid.js rm dupli

EHIS03042 committed:
Delete backend/backend/scripts/fixSpecificFlowers.js rm dupli

EHIS03042 committed:
Delete backend/backend/scripts/migrateImages.js rm dupli

EHIS03042 committed:
Delete backend/backend/scripts/migrateImagesOptimized.js rm dupli

EHIS03042 committed:
Delete backend/backend/scripts/normalizeCategories.js rm dupli

EHIS03042 committed:
Delete backend/backend/fixImagePaths.js rm dupli

EHIS03042 committed:
Delete backend/backend/LICENSE rm dupli

EHIS03042 committed:
Delete backend/backend/package-lock.json rm dupli

EHIS03042 committed:
Delete backend/backend/package.json rm dupli

EHIS03042 committed:
Delete backend/backend/server.js rm dupli

EHIS03042 committed:
backend
